service/history/workflowExecutionContext.go

```diff
@@ -1127,3 +1169,59 @@ func (c *workflowExecutionContextImpl) resetWorkflowExecution(
 func (c *workflowExecutionContextImpl) getQueryRegistry() QueryRegistry {
 	return c.queryRegistry
 }
+
+func (c *workflowExecutionContextImpl) reapplyEvents(
+	ctx context.Context,
+	events *persistence.WorkflowEvents,
+) error {
+
+	reapplyEvents := []*workflow.HistoryEvent{}
+	// TODO: need to implement Reapply policy
+	for _, event := range events.Events {
+		switch event.GetEventType() {
+		case workflow.EventTypeWorkflowExecutionSignaled:
+			reapplyEvents = append(reapplyEvents, event)
+		}
+	}
+	// there is no event to reapply
+	if len(reapplyEvents) == 0 {
+		return nil
+	}
+
+	domainID := events.DomainID
+	execution := &workflow.WorkflowExecution{
+		WorkflowId: common.StringPtr(events.WorkflowID),
```

Comment:
could you add a comment here specifying that run ID must NOT be set

Reply:
done